fractions/javaee/jaxrs/src/main/java/org/wildfly/swarm/jaxrs/internal/JAXRSArchiveImpl.java

```diff
@@ -99,8 +112,62 @@ private static boolean hasApplicationPathAnnotation(Archive<?> archive) {
         return false;
     }
 
+    private static boolean hasApplicationServletMapping(Archive<?> archive) {
+        Node webXmlNode = archive.get(PATH_WEB_XML);
+        if (webXmlNode != null) {
+            return hasApplicationServletMapping(webXmlNode.getAsset());
+        }
+        return false;
+    }
+
+    private static boolean hasApplicationServletMapping(Asset webXmlAsset) {
```

Comment:
Instead of this XML parsing and XPath, can you use the WebXmlAsset directly?

https://github.com/wildfly-swarm/wildfly-swarm/blob/master/fractions/javaee/undertow/src/main/java/org/wildfly/swarm/undertow/descriptors/WebXmlAsset.java#L80

Or add more accessors if we you need direct access to the underlying `WebAppDescriptor` which should already fully model the `web.xml`?